src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -288,15 +288,19 @@
             gc-interval-millis (to-millis gc-interval)
             dlo-compression-interval-millis (to-millis dlo-compression-interval)
             seconds-pos? (comp pos? to-seconds)
-            db-maintenance-tasks (fn []
-                                   (do
-                                     (when (seconds-pos? node-ttl) (auto-expire-nodes! node-ttl write-db))
-                                     (when (seconds-pos? node-purge-ttl) (purge-nodes! node-purge-ttl write-db))
-                                     (when (seconds-pos? report-ttl) (sweep-reports! report-ttl write-db))
-                                     ;; Order is important here to ensure
-                                     ;; anything referencing an env or resource
-                                     ;; param is purged first
-                                     (garbage-collect! write-db)))]
+            db-maintenance-tasks #(try
+                                    (when (seconds-pos? node-ttl)
+                                      (auto-expire-nodes! node-ttl write-db))
+                                    (when (seconds-pos? node-purge-ttl)
+                                      (purge-nodes! node-purge-ttl write-db))
+                                    (when (seconds-pos? report-ttl)
+                                      (sweep-reports! report-ttl write-db))
+                                    ;; Order is important here to ensure
+                                    ;; anything referencing an env or resource
+                                    ;; param is purged first
+                                    (garbage-collect! write-db)
+                                    (catch Exception ex
+                                      (log/error ex)))]
```

Comment:
Do we really want to swallow the exception here and not rethrow it? Today at-at doesn't do error handling properly, so the exception gets ignored, but do you expect it to always be that way?